app/models/dossier.rb

```diff
@@ -345,26 +345,23 @@ def classer_sans_suite(motivation: nil, instructeur: nil, processed_at: Time.zon
   scope :without_en_construction_expiration_notice_sent, -> { where(en_construction_close_to_expiration_notice_sent_at: nil) }
   scope :without_termine_expiration_notice_sent, -> { where(termine_close_to_expiration_notice_sent_at: nil) }
 
-  scope :discarded_expired, -> { discarded.where('dossiers.hidden_at < ?', 1.week.ago) }
-  scope :discarded_by_user_expired, -> { discarded.where('dossiers.hidden_by_user_at < ?', 1.week.ago) }
-  scope :discarded_by_administration_expired, -> { discarded.where('dossiers.hidden_by_administration_at < ?', 1.week.ago) }
-  scope :discarded_brouillon_expired, -> do
-    with_discarded
-      .state_brouillon
-      .discarded_expired
-      .or(state_brouillon.discarded_by_user_expired)
-  end
-  scope :discarded_en_construction_expired, -> do
-    with_discarded
-      .state_en_construction
-      .discarded_expired
-      .or(state_en_construction.discarded_by_user_expired)
-  end
-  scope :discarded_termine_expired, -> do
-    with_discarded
-      .state_termine
-      .discarded_expired
-      .or(state_termine.discarded_by_user_expired.discarded_by_administration_expired)
+  scope :expired, -> { where('dossiers.hidden_at < ?', 1.week.ago) }
```

Comment:
Si j'ai bien tout compris, on garde ce scope pour gérer les anciens fichiers cachés. c'est ca ?

Est ce qu'on devrait pas mieux faire une migration du genre `Dossier.discarded.update_all('hidden_by_user_at = hidden_at and hidden_by_administration_at = hidden_at)` et dégager le scope expired ?

J'ai l'impression que sinon ça impose aux forks un délai d'une semaine avant de pouvoir retier la colonne `hidden_at`

Reply:
Effectivement, on pourrait faire une migration comme proposée. Je ne suis pas 100% convaincue par la query de Sim, mais l'idée est là.

Reply:
Oui c'est ça @LeSim 

Le truc avec la migration c'est qu'on utilise plus discarded du coup ça serait :

`Dossier.where.not(hidden_at: nil).update_all('hidden_by_user_at = hidden_at and hidden_by_administration_at = hidden_at)`

cc @tchak 

Mais perso je suis plutôt pour le faire en plusierus fois et attendre une semaine, au moins si on a des merdes ou si on doit rollback pour une raison ou une autre on aura pas touché à la base

Reply:
vu avec @tchak on code la migration de la donnée dans une autre pr